com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/util/ParseTreeUtil.java

```diff
@@ -138,8 +138,11 @@ public static String getParsedString(final EObject object, final String featureN
   public static void ensureNodeModelLoaded(final Resource resource) {
     if (resource instanceof LazyLinkingResource2 lazyLinkinResource && lazyLinkinResource.isLoadedFromStorage()) {
       EObject root = resource.getContents().get(0);
-      if (root != null && NodeModelUtils.getNode(root) instanceof ICompositeNode composite) {

```

Comment:
Your version is fine for me, but what about just naming the variable `rootNode` instead of `composite`?  That would be a smaller change than this one.